### PR TITLE
Change `data-type` to `datatype`

### LINKED ISSUE #169: change data-type to datatype in docs
This also requires changing the CLI argument `--data-type` to `--datatype`.

### PR DESCRIPTION
**Pull request by JoeZiminski**
closes #169 

This PR changes the formatting of `data-type` to `datatype` for alignment with NeuroBlueprint. This was a bit more painful that I anticipated, in addition to changing

1) A CLI argument (`--data-type`)
2) Use of "data-type" in documentation.
3) references to "data-type" in code dostrings.

`data_type` was used as a variable name all over the place, which caused some problems with the CLI - API link as API should mirror CLI. However, the replacing seems to have gone well as all tests pass and documentation looks okay.

### CODE REVIEW

File: docs/source/pages/documentation.md [hunk 1]
```diff
@@ -182,7 +182,7 @@ upload \
 
 Will *upload* (from *local* to *central* ) _behavioural_ _sessions_ 5 and 6, collected at any date, for _subjects_ 1 to 3.
 
-The keyword `all` can be input in place of a `-sub`, `-ses` or _data-type_ argument `dt` to transfer all available subject, sessions or data types available. For example:
+The keyword `all` can be input in place of a `-sub`, `-ses` or _datatype_ argument `dt` to transfer all available subject, sessions or data types available. For example:
```
Comment (JoeZiminski) [inline docs/source/pages/documentation.md:185]:
```suggestion
The keyword `all` can be input in place of a `-sub`, `-ses` or _datatype_ argument `-dt` to transfer all available subject, sessions or data types available. For example:
```

File: docs/source/pages/documentation.md [hunk 2]
```diff
@@ -510,14 +510,14 @@ DataShuttle provides a number of keyword arguments to allow separate handling of
...
-`all` : All *data-type* folders at the *subject* or *session* folder level will be transferred, as well as all files and folders within selected *session* folders.
+`all` : All *datatype* folders at the *subject* or *session* folder level will be transferred, as well as all files and folders within selected *session* folders.
 
-`all_data_type` : All *data-type* folders (i.e. folders with the pre-determined name: `behav`, `ephys`, `funcimg`, `histology`) residing at either the *subject* or *session* level will be
-transferred. Non-*data-type* folders at the *session* level will not be transferred
+`all_data_type` : All *datatype* folders (i.e. folders with the pre-determined name: `behav`, `ephys`, `funcimg`, `histology`) residing at either the *subject* or *session* level will be
+transferred. Non-*datatype* folders at the *session* level will not be transferred
 
-`all_ses_level_non_data_type` : Non *-data-type* folders at the *session* level will not be transferred
+`all_ses_level_non_data_type` : Non *-datatype* folders at the *session* level will not be transferred
```
Comment (JoeZiminski) [inline docs/source/pages/documentation.md:520]:
```suggestion
`all_ses_level_non_data_type` : Non *datatype* folders at the *session* level will not be transferred
```

**Review by niksirbi — APPROVED** [review]
Looks good @JoeZiminski. All the tests pass locally as well and the documentation build still works.
I only found one lingering mention of "data types" in README.md (en of line 11). I leave it up to you whether you want to make that into "datatypes" or not.

**Comment by JoeZiminski** [comment]
Thanks @niksirbi !

**Pull request merged**